python/paddle/device/cuda/__init__.py

```diff
@@ -204,3 +207,63 @@ def stream_guard(stream):
             yield
         finally:
             stream = _set_current_stream(pre_stream)
+
+
+def get_device_properties(device=None):
+    '''
+    Return the properties of given CUDA device.
+
+    Args:
+        device(paddle.CUDAPlace() or int or str): The device, the ID of the device 
+            or the string name of device like 'gpu:x' which to get the properties 
+            of the device from. If device is None, the device is the current device. 
+            Default: None.
+
+    Returns:
+        _CudaDeviceProperties: the properties of the device which include ASCII string 
+            identifying device, major compute capability, minor compute capability, global 
+            memory available on device in bytes and the number of multiprocessors on the device.
+
+    Examples:
+    
+        .. code-block:: python
+
+            # required: gpu
+
+            import paddle
+            paddle.device.cuda.get_device_properties()
+            paddle.device.cuda.get_device_properties(0)
+            paddle.device.cuda.get_device_properties('gpu:0')
+            paddle.device.cuda.get_device_properties(paddle.CUDAPlace(0))
+
+    '''
+
+    place = framework._current_expected_place()
+    if not isinstance(place, core.CUDAPlace) or not is_compiled_with_cuda():
+        raise ValueError(
+            "The current device: {} is not expected. Because paddle.device.cuda."
+            "get_device_properties only support cuda device Please change device"
+            "and input device again!".format(place))
```

Comment:
I am confused with this sentence. What are the differences between `device` and `input device`?